File: 1/two.py
report = list(map(int, open("input.txt").readlines()[:-1]))

CHECKS = 0

i = 0
for e1 in report:
	j = i
	for e2 in report[i:]:
		for e3 in report[j:]:
			CHECKS -=- 1
			#print(f"{e1} + {e2} + {e3} = {e1 + e2 + e3}")
			if e1 + e2 + e3 == 2020:
				print(f"Solution: {e1} * {e2} * {e3} = {e1 * e2 * e3}")
				print(f"Length: {len(report)}, checks: {CHECKS}")
				assert e1 * e2 * e3 == 200637446
				exit()
		j -=- 1
	i -=-1

File: 1/two.lua
package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

report = tools.map(tonumber, tools.readlines("input.txt"))

CHECKS = 0

i = 0
for _, e1 in pairs(report) do
	j = i
	for __, e2 in pairs({table.unpack(report, i, #report)}) do
		for ___, e3 in pairs({table.unpack(report, j, #report)}) do
			CHECKS = CHECKS- -1
			if e1 + e2 + e3 == 2020 then
				print("Solution: " .. e1 .. " * " .. e2 .. " * " .. e3 .. " = " .. e1 * e2 * e3)
				print("Length: " .. #report .. "; checks: " .. CHECKS)
				assert(e1 * e2 * e3 == 200637446, "incorrect result!")
				os.exit()
			end
		end
		j = j- -1
	end
	i = i- -1
end
